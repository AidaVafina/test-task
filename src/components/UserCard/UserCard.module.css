.card {
    background: white;
    border-radius: 8px;
    padding: 1.5rem;
    box-shadow: 0 2px 4px rgba(0,0,0,0.1);
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    cursor: pointer;
    -webkit-font-smoothing: antialiased;
    transform: translate3d(0, 0, 0);
    transition: transform 0.3s cubic-bezier(0.4, 0, 0.2, 1);
}

.card:hover {
    transform: translate3d(0, -2px, 0);
}

.name {
    font-size: 1.2rem;
    font-weight: 600;
    color: var(--color-text-primary, #333);
    margin: 0;
}

.email {
    color: var(--color-text-secondary, #666);
    margin: 0;
    font-size: 0.9rem;
}

.city {
    color: var(--color-text-secondary, #666);
    margin: 0;
    font-size: 0.9rem;
}

.link {
    display: inline-block;
    margin-top: 1rem;
    padding: 0.5rem 1rem;
    background-color: var(--color-primary, #3498db);
    color: white;
    text-decoration: none;
    border-radius: 4px;
    font-size: 0.9rem;
    text-align: center;
    transition: background-color 0.2s ease;
}

.link:hover {
    background-color: var(--color-primary-dark, #2980b9);
}
