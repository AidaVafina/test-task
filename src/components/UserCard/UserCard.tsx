import React from 'react';
import styles from './UserCard.module.css';
import { Link } from 'react-router-dom';
import type { User } from "../../types";
import { motion } from 'framer-motion';

interface UserCardProps {
    user: User;
    index: number;
}

const UserCard: React.FC<UserCardProps> = ({ user, index }) => {
    return (
        <motion.div
            className={styles.card}
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
            transition={{
                duration: 0.3,
                delay: index * 0.1, 
                ease: "easeOut"
            }}
            whileHover={{
                scale: 1.05,
                boxShadow: "0 4px 8px rgba(0,0,0,0.1)",
                transition: { duration: 0.2 }
            }}
        >
            <motion.h3 
                className={styles.name}
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                transition={{ delay: 0.2 + index * 0.1 }}
            >
                {user.name}
            </motion.h3>
            <motion.p 
                className={styles.email}
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                transition={{ delay: 0.3 + index * 0.1 }}
            >
                {user.email}
            </motion.p>
            <motion.p 
                className={styles.city}
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                transition={{ delay: 0.4 + index * 0.1 }}
            >
                {user.address.city}
            </motion.p>
            <motion.div
                whileHover={{ scale: 1.1 }}
                whileTap={{ scale: 0.95 }}
            >
                <Link to={`/user/${user.id}`} className={styles.link}>
                    Подробнее
                </Link>
            </motion.div>
        </motion.div>
    );
};

export default UserCard;
