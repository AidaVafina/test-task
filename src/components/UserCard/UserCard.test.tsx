import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import UserCard from './UserCard';

// моковые данные для теста
const mockUser = {
    id: 1,
    name: 'John Doe',
    username: 'johndoe',
    email: 'john@example.com',
    address: {
        city: 'New York',
        street: '123 Main St',
        suite: 'Apt 4B',
        zipcode: '10001',
        geo: {
            lat: '40.7128',
            lng: '-74.0060'
        }
    },
    phone: '123-456-7890',
    website: 'johndoe.com',
    company: {
        name: 'Test Company',
        catchPhrase: 'Test catch phrase',
        bs: 'Test bs'
    }
};

describe('UserCard Component', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <BrowserRouter>
                <UserCard user={mockUser} index={0} />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render user information correctly', () => {
        const { container } = render(
            <BrowserRouter>
                <UserCard user={mockUser} index={0} />
            </BrowserRouter>
        );

        expect(container.textContent).toContain(mockUser.name);
        expect(container.textContent).toContain(mockUser.email);
        expect(container.textContent).toContain(mockUser.address.city);
        expect(container.textContent).toContain('Подробнее');
    });
}); 