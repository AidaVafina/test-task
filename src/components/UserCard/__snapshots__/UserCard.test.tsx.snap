// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserCard Component > should match snapshot 1`] = `
<div>
  <div
    class="_card_c4bfad"
    style="opacity: 0; transform: translateY(20px);"
  >
    <h3
      class="_name_c4bfad"
      style="opacity: 0;"
    >
      John Doe
    </h3>
    <p
      class="_email_c4bfad"
      style="opacity: 0;"
    >
      john@example.com
    </p>
    <p
      class="_city_c4bfad"
      style="opacity: 0;"
    >
      New York
    </p>
    <div
      tabindex="0"
    >
      <a
        class="_link_c4bfad"
        data-discover="true"
        href="/user/1"
      >
        Подробнее
      </a>
    </div>
  </div>
</div>
`;

exports[`UserCard Component should match snapshot 1`] = `
<div>
  <div
    class="card"
    style="opacity: 0; transform: translateY(20px);"
  >
    <h3
      class="name"
      style="opacity: 0;"
    >
      John Doe
    </h3>
    <p
      class="email"
      style="opacity: 0;"
    >
      john@example.com
    </p>
    <p
      class="city"
      style="opacity: 0;"
    >
      New York
    </p>
    <div
      tabindex="0"
    >
      <a
        class="link"
        data-discover="true"
        href="/user/1"
      >
        Подробнее
      </a>
    </div>
  </div>
</div>
`;
